### DISCUSSION TITLE
Can JSqlParser be used commercially for free of cost?

### DISCUSSION THREAD

**Discussion by softKarma** [JSQLParser/JSqlParser#1894] [Q&A] [answered]
I am a bit confused about the license as it is not just Apache 2.0 but both Apache-2.0 and LGPL-2.1 licenses. As par as I know a library LGPL-2.1 cannot be used freely (not only in form of money) commercially but Apache-2.0 can be. So which one is it?

**Comment by manticore-projects** [comment, marked as answer]
Greetings.

You can use `JSQLParser` either under the terms of the `Apache 2.0 License` **OR** under the terms of the `LGPL-2.1` because both licenses have been granted. If you use it under  the  `Apache 2.0 License` then you are not bound to the terms of the `LGPL-2.1`.

Although from the business ethical point of view, any commercial use should warrant the company's or entrepreneurs appreciation and financial contribution to the `JSQLParser` project. You will sleep much better, trust me.

**Reply by softKarma** [reply]
I agree. I was asking more from a legal point of view. Thanks for the confirmation.
